packages/server/src/next-utils.ts

```diff
@@ -317,9 +338,6 @@ export function startCustomServer(
       }
     }
 
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    esbuild.build(esbuildOptions).then(() => {
-      spawnServer()
-    })
+    spawnServer()
```

Comment:
Looking good! This is a regression for `blitz dev` use case.

Here's the full list. Sorry, I should have put this earlier.

- [x] `blitz build` builds custom server
- [x] `blitz start` runs but does not build custom server
- [ ] `blitz dev` should both build and run custom server (this is working correctly in existing code, but it's not on this PR)

Reply:
I think adding:
`await buildCustomServer()` to this line will solve it:
https://github.com/blitz-js/blitz/blob/8eb1509b9ad45b647a702381b4f693ed3b8aefe6/packages/server/src/dev.ts#L48

Reply:
That handles initial build, but not watch. Did you test `blitz dev`?

I think we should conditionally do `spawnServer()` vs 

```
esbuild.build(esbuildOptions).then(() => {
      spawnServer()
})
```

based on watch flag. Currently `esbuildOptions` in this function are not used and so hot reload doesn't work.

Reply:
You're right, I rushed this one in. This is what I am coming up with now.

```javascript

export function startCustomServer(
    cwd: string,
    config: ServerConfig,
    {watch}: CustomServerOptions = {},
) {
  const serverBuildPath = getCustomServerBuildPath()

  let spawnEnv = getSpawnEnv(config)
  if (config.env === "prod") {
    spawnEnv = {...spawnEnv, NODE_ENV: "production"}
  }

  return new Promise<void>((res, rej) => {
    let process: ChildProcess

    const RESTART_CODE = 777777

    const spawnServer = () => {
      process = spawn("node", [serverBuildPath], {
        cwd,
        env: spawnEnv,
        stdio: "inherit",
      })
          .on("exit", (code: number) => {
            if (code === 0) {
              res()
            } else if (watch && code === RESTART_CODE) {
              spawnServer()
            } else {
              rej(`server.js failed with status code: ${code}`)
            }
          })
          .on("error", (err) => {
            console.error(err)
            rej(err)
          })
    }

    if (watch) {
      // Handle build & Starting server
      const esbuildOptions = getEsbuildOptions()
      esbuildOptions.watch = {
        onRebuild(error) {
          if (error) {
            log.error("Failed to re-build custom server")
          } else {
            log.newline()
            log.progress("Custom server changed - restarting...")
            log.newline()
            //@ts-ignore -- incorrect TS type from node
            process.exitCode = RESTART_CODE
            process.kill("SIGABRT")
          }
        },
      }
      // eslint-disable-next-line @typescript-eslint/no-floating-promises
      esbuild.build(esbuildOptions).then(() => {
        spawnServer()
      })
    } else {
      // No build required, Start server
      spawnServer()
    }
  })
}
```